packages/tools/fetch-tool/src/fluidFetchMessages.ts

```diff
@@ -145,11 +176,12 @@ async function* saveOps(
     if (files.length !== 0) {
         index = (files.length - 1);
     }
-    let curr = index * chunk + 1;
+    let curr = index * chunk + currSeq;
```

Comment:
You set the value the currSeq to last message stored in files or the firstAvailableDelta. I don't think this calculation is correct. Should the previous logic work here? Can you explain what you are trying to do here?

Reply:
i'm setting it to firstAvailableDelta, if the firstAvailableDelta is not 1, then the assert will fail in line 197, but shouldn't the `curr` be the sequence number of first message to save? just like the assert. and the first message has firstAvailableDelta as the sequence number instead of 1

Reply:
Yes.

Reply:
I think what we doing here is to start writing the content to the last file. So lets say the ops fetched on last try was up to 250K ops and we are storing 100K ops in 1 file as per the logic. So lets say on next try we have 20K more ops for the same document. Then we want to write the last 20K ops to the last file and not create a new file for the same as 20K ops will fit in same file. 
So we want to preserve same logic. 
Here you just need to make sure that you are storing 100K ops per file.

Reply:
yes I'm still storing 100k ops per file, in line 215 `curr` will increment by chunk. but to start, `curr` can't just be 1, because the first available sequence number might not be 1, it might be the firstAvailableDelta

Reply:
I think there are these scenarios:
1. when there is no message.json, and there is no gap, curr will be 1.
2. when there is no message.json, but there is gap in ops, curr will be firstAvailableDelta
3. when there is existing message.json, and there is gap, there will be an error to tell user to delete it, so it will go back to scenario 1.
4. when there is existing message.json, there is no gap, curr will be the index*chunk + this 
![image](https://user-images.githubusercontent.com/34214774/132454699-1a3bf81d-79dc-4d2c-81cd-54dc49176c99.png)

Reply:
i guess i did not include the scenario when there are multiple messages.json files. so if index is not zero, then curr should be sequence number from last file and add 1.

Reply:
wait, even if there are multiple files, and there are gaps, we would still tell the user to delete the files and fetch again. and we will fetch from firstAvailableDelta.

Reply:
fixed now